### PR TITLE
Add stats for Workers in queue

### PR DESCRIPTION
**Pull request by naduvima**
Increment and decrements workers count  when enqueued, dequeued and stored in REDIS. 
Emits the  count through worker.

How to Enable?
```
          Sidekiq::Instrument::WorkerMetrics.enabled = true
          Sidekiq::Instrument::WorkerMetrics.redis_config = {
            host:        ENV['REDIS_HOST'],
            port:        ENV['REDIS_PORT'],
            db:          0
          }
  ```

### CODE REVIEW

File: sidekiq-instrument.gemspec [hunk 1]
```diff
@@ -25,6 +25,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake', '~> 12.0'
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'pry-byebug', '~> 3.4'
+  spec.add_dependency "redis-client", ">= 0.11.0"
```
Comment (hkim3162) [inline sidekiq-instrument.gemspec]: small coment: please move to L23

File: lib/sidekiq/instrument/middleware/client.rb [hunk 2]
```diff
@@ -1,4 +1,7 @@
+# frozen_string_literal: true
+
 require 'sidekiq/instrument/mixin'
+require 'pry'
```
Comment (hkim3162) [inline lib/sidekiq/instrument/middleware/client.rb]: We can remove the pry's if no longer used

File: lib/sidekiq/instrument/middleware/server.rb [hunk 3]
```diff
@@ -9,6 +12,7 @@ def call(worker, job, queue, &block)
       Statter.dogstatsd&.increment('sidekiq.dequeue', worker_dog_options(worker))
 
       start_time = Time.now
+      WorkerMetrics.trace_workers_decrement_counter(worker.class.to_s)
```
Comment (hkim3162) [inline lib/sidekiq/instrument/middleware/server.rb]: does this need to be downcased?

File: spec/sidekiq-instrument/client_middleware_spec.rb [hunk 4]
```diff
@@ -39,6 +39,39 @@
       end
     end
 
+    context 'with WorkerMetrics.enabled true' do
```
Comment (hkim3162) [inline spec/sidekiq-instrument/client_middleware_spec.rb:42]: Can we create a separate spec file for`worker_metrics` and unit tests?

File: lib/sidekiq/instrument/worker_metrics.rb [hunk 5]
```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'redis'
+require 'redis-client'
+module Sidekiq
+  module Instrument
+    # Stores worker count
```
Comment (hkim3162) [inline lib/sidekiq/instrument/worker_metrics.rb]: Please provide a little more context on the main benefit of this.

File: spec/sidekiq-instrument/client_middleware_spec.rb (hunk 4 above)
Comment (naduvima) [reply spec/sidekiq-instrument/client_middleware_spec.rb:42]: not much scope to move to another file, scenarios can be tested with same context.

**Review by hkim3162 — DISMISSED** [review]

**Comment by hkim3162** [comment]
Please reference testing of this manually while referencing this gem version in portfolio

**Review by SheffAtEnova — APPROVED** [review]

**Review by SheffAtEnova — APPROVED** [review]

**Pull request merged**